radixdlt/src/integration/java/com/radixdlt/integration/recovery/OneNodeAlwaysAliveSafetyTest.java

```diff
@@ -0,0 +1,255 @@
+
+		for (Supplier<Injector> nodeCreator : nodeCreators) {
+			this.nodes.add(nodeCreator.get());
+		}
+	}
+
+	private Injector createRunner(ECKeyPair ecKeyPair, List<BFTNode> allNodes) {
+		final BFTNode self = BFTNode.create(ecKeyPair.getPublicKey());
+
+		return Guice.createInjector(
+			ModuleForRecoveryTests.create(),
+			new AbstractModule() {
+
+				@ProvidesIntoSet
+				@ProcessOnDispatch
+				private EventProcessor<BFTCommittedUpdate> committedUpdateEventProcessor(@Self BFTNode node) {
+					return nodeEvents.processor(node, BFTCommittedUpdate.class);
+				}
+
+				@ProvidesIntoSet
+				@ProcessOnDispatch
+				private EventProcessor<FormedQC> formedQCEventProcessor(@Self BFTNode node) {
+					return nodeEvents.processor(node, FormedQC.class);
+				}
+
+				@Override
+				protected void configure() {
+					bind(HashSigner.class).toInstance(ecKeyPair::sign);
+					bind(BFTNode.class).annotatedWith(Self.class).toInstance(self);
+					bind(new TypeLiteral<List<BFTNode>>() { }).toInstance(allNodes);
+					bind(ControlledSenderFactory.class).toInstance(network::createSender);
+					bind(View.class).annotatedWith(EpochCeilingView.class).toInstance(View.of(88L));
+
+					final RuntimeProperties runtimeProperties;
+					// TODO: this constructor/class/inheritance/dependency is horribly broken
+					try {
+						runtimeProperties = new RuntimeProperties(new JSONObject(), new String[0]);
+						runtimeProperties.set("db.location", folder.getRoot().getAbsolutePath() + "/RADIXDB_RECOVERY_TEST_" + self);
+					} catch (ParseException e) {
+						throw new IllegalStateException();
+					}
+					bind(RuntimeProperties.class).toInstance(runtimeProperties);
+				}
+			}
+		);
+	}
+
+	private void restartNode(int index) {
+		this.network.dropMessages(m -> m.channelId().receiverIndex() == index);
+		Injector injector = nodeCreators.get(index).get();
+		this.nodes.set(index, injector);
+	}
+
+	private void startNode(int index) {
+		Injector injector = nodes.get(index);
+		String bftNode = " " + injector.getInstance(Key.get(BFTNode.class, Self.class));
+		ThreadContext.put("bftNode", bftNode);
+		try {
+			injector.getInstance(DeterministicEpochsConsensusProcessor.class).start();
+		} finally {
+			ThreadContext.remove("bftNode");
+		}
+	}
+
+	private void processUntilNextCommittedUpdate() {
+		lastNodeToCommit = -1;
+
+		while (lastNodeToCommit == -1) {
+			Timed<ControlledMessage> msg = this.network.nextMessage();
+			logger.debug("Processing message {}", msg);
+
+			int nodeIndex = msg.value().channelId().receiverIndex();
+			Injector injector = this.nodes.get(nodeIndex);
+			String bftNode = " " + injector.getInstance(Key.get(BFTNode.class, Self.class));
+			ThreadContext.put("bftNode", bftNode);
+			try {
+				injector.getInstance(DeterministicEpochsConsensusProcessor.class).handleMessage(msg.value().origin(), msg.value().message());
+			} finally {
+				ThreadContext.remove("bftNode");
+			}
+		}
+	}
+
+	@Test
+	@Ignore("Safety broken here.")
```

Comment:
Can we get an issue in Jira, and a reference to that issue in the comment here please.